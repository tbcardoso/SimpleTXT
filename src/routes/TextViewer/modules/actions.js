/*
 * action types
 */

export const SELECT_FILE_SUCCESS = 'SELECT_FILE_SUCCESS'
export const LOAD_FILE_SUCCESS = 'LOAD_FILE_SUCCESS'



/*
 * action creators
 */

export function selectFileSuccess(filePath) {
  return { type: SELECT_FILE_SUCCESS, filePath }
}

export function loadFileSuccess(filePath, fileContent) {
  return { type: LOAD_FILE_SUCCESS, filePath, fileContent }
}