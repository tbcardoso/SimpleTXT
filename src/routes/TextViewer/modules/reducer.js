import { combineReducers } from 'redux';
import * as Actions from './actions';
import about from '../../../assets/about';

const selectedFileInitialState = {
	filePath: '',
	fileContent: '',
	loadingFilePath: '',
	error: '',
};

function selectedFile(state = selectedFileInitialState, action) {
	switch (action.type) {
		case Actions.SELECT_FILE_SUCCESS:
			return { ...state, loadingFilePath: action.filePath };
		case Actions.LOAD_FILE_SUCCESS:
			return { ...state, loadingFilePath: '', filePath: action.filePath, fileContent: action.fileContent, error: '' };
		case Actions.LOAD_FILE_ERROR:
			return { ...state, loadingFilePath: '', error: action.error };
		case Actions.ABOUT_SCREEN_SELECTED:
			return { ...state, loadingFilePath: '', filePath: 'About', fileContent: about, error: '' };
		default:
			return state;
	}
}

export default combineReducers({
	selectedFile,
});
