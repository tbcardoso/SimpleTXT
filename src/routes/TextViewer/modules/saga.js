import { call, put, takeLatest } from 'redux-saga/effects';
import * as Actions from './actions';
import RNFetchBlob from 'react-native-fetch-blob';
import { ToastAndroid } from 'react-native';

function * selectedFileSaga(action) {
	console.log('Selected file Saga: ', action);

	try {
		const data = yield call(RNFetchBlob.fs.readFile, action.filePath, 'utf8');
		//console.log('File data: ', data);
		yield put(Actions.loadFileSuccess(action.filePath, data));
	} catch (e) {
		console.log('Load File Error: ', e);
		ToastAndroid.show(e.message, ToastAndroid.SHORT)
		yield put(Actions.loadFileError(action.filePath, e.message));
	}
}

export default function * saga() {
	yield takeLatest(Actions.SELECT_FILE_SUCCESS, selectedFileSaga);
}
