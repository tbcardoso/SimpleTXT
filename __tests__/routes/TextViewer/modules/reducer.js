/* eslint-env jest */
import * as Actions from '../../../../src/routes/TextViewer/modules/actions';
import { selectedFileInitialState, selectedFile } from '../../../../src/routes/TextViewer/modules/reducer';

describe('Reducer selectedFile', () => {
	test('Action selectFileSuccess', () => {
		expect(selectedFile(selectedFileInitialState, Actions.selectFileSuccess('testPath'))).toMatchSnapshot();
	});

	test('Action loadFileSuccess', () => {
		expect(selectedFile(selectedFileInitialState, Actions.loadFileSuccess('testPath', 'Lorem Ipsum...'))).toMatchSnapshot();
	});

	test('Action loadFileError', () => {
		expect(selectedFile(selectedFileInitialState, Actions.loadFileError('testPath', 'Error message.'))).toMatchSnapshot();
	});

	test('Action aboutScreenSelected', () => {
		expect(selectedFile(selectedFileInitialState, Actions.aboutScreenSelected())).toMatchSnapshot();
	});
});
