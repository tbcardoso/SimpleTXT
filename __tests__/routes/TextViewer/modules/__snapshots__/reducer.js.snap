// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer selectedFile Action aboutScreenSelected 1`] = `
Object {
  "error": "",
  "fileContent": "SimpleTXT
	Created by Tiago Cardoso
	Licensed under the GNU General Public License (GPLv3)


Used libraries and assets

	React, by Facebook, Inc., licensed under the BSD License
	React Native, by Facebook, Inc., licensed under the BSD License
	realm, by Realm, licensed under the Apache License Version 2.0
	Redux, by Dan Abramov, licensed under the MIT License
	react-redux, by Dan Abramov, licensed under the MIT License
	redux-saga, by Yassine Elouafi, licensed under the MIT License
	react-native-fetch-blob, licensed under the MIT License
	react-native-file-picker
	react-native-vector-icons, by Joel Arvidsson, licensed under the MIT License
	Material Icons, by Google, licensed under the Apache License Version 2.0
	Launcher Icon made by Dimitry Miroliubov from www.flaticon.com, licensed under the Flaticon Basic License
",
  "filePath": "About",
  "loadingFilePath": "",
}
`;

exports[`Reducer selectedFile Action loadFileError 1`] = `
Object {
  "error": "Error message.",
  "fileContent": "",
  "filePath": "",
  "loadingFilePath": "",
}
`;

exports[`Reducer selectedFile Action loadFileSuccess 1`] = `
Object {
  "error": "",
  "fileContent": "Lorem Ipsum...",
  "filePath": "testPath",
  "loadingFilePath": "",
}
`;

exports[`Reducer selectedFile Action selectFileSuccess 1`] = `
Object {
  "error": "",
  "fileContent": "",
  "filePath": "",
  "loadingFilePath": "testPath",
}
`;
